src/App.tsx

```diff
@@ -2,10 +2,54 @@ import React from 'react'
 import logo from './logo.svg'
 import './App.css'
 import FileImportButton from './Components/FileImportButton'
+import TextField from '@material-ui/core/TextField'
+import xml2js from 'xml2js'
+import Attempt from './types'
 
 const App = () => {
-  function handleFile() {
-    console.log('test')
+  const [attemptHistory, setAttemptHistory] = React.useState<Attempt[]>([])
+
+  function handleFile(e: React.ChangeEvent<HTMLInputElement>) {
+    if (e && e.target && e.target.files && e.target.files[0]) {
+      const file = e.target.files[0]
+
+      let reader = new FileReader()
```

Comment:
const

Reply:
Unless you can't assign to a const's properties, like you do on line 17. Not actually sure about that.

Reply:
I assumed I couldn't, but it still seems to work with a const so I'll change it

Reply:
Yeah, consts are a little nuanced like that. It's only the top level reference that cant change, properties, etc. can. There are some cases where the intention of changing properties might  lead you to use `let` anyway. But in cases like this where the DX of using a FileReader involves setting a method I don't really consider that to be "modifying" the variable.